src/Controller/CRUDController.php

```diff
@@ -1008,22 +1101,14 @@ protected function configure(): void
         }
 
         try {
-            $this->admin = $this->container->get('sonata.admin.pool')->getAdminByAdminCode($adminCode);
+            $this->admin = $this->pool->getAdminByAdminCode($adminCode);
         } catch (\InvalidArgumentException $e) {
             throw new \RuntimeException(sprintf(
                 'Unable to find the admin class related to the current controller (%s)',
                 static::class
             ));
         }
 
-        $this->templateRegistry = $this->container->get(sprintf('%s.template_registry', $this->admin->getCode()));
```

Comment:
This change is not correct, you cant replace the specific admin templater registry with the generic one (as explained in the previous pr

Reply:
Can you please help me with this one ?

Reply:
https://github.com/sonata-project/SonataAdminBundle/pull/6387

After this change(it must be merge to master too) you will be able to get `TemplateRegister` directly from `Admin`.

Reply:
@mathieupetrini #6387 is megre in master now. Can you finish your work now?

Reply:
```suggestion
     $this->templateRegistry = $this->admin->getTemplateRegistry();
```